libevm/VM.cpp

```diff
@@ -87,15 +86,31 @@ void VM::onOperation()
 			m_newMemSize > m_mem.size() ? (m_newMemSize - m_mem.size()) / 32 : uint64_t(0),
 			m_runGas, m_io_gas, this, m_ext);
 }
+#if EVM_HACK_ON_OPERATION
+	#define onOperation()
+#endif
 
-void VM::checkStack(unsigned _removed, unsigned _added)
+//
+// set current SP to SP', adjust SP' per _removed and _added items
+//
+void VM::adjustStack(unsigned _removed, unsigned _added)
```

Comment:
I'll look at it.  I think the `unsigned` types go back to the older code this came from, and they get passed to throwBadStack() function which expects unsigned.  Overall I don't think the code is always using the optimal data types.  (Not sure I understand the pessimization problem - _removed and _added are not going to overflow/)